ex2_services/Services/source/adcs/adcs_service.c

```diff
@@ -1251,9 +1260,112 @@ SAT_returnState adcs_service_app(csp_packet_t *packet) {
     }
 
     case ADCS_SET_CUBESENSE_CONFIG: {
-        cubesense_config config;
-        memcpy(&config, &packet->data[IN_DATA_BYTE], sizeof(cubesense_config));
-        status = ADCS_set_cubesense_config(config);
+        cubesense_config cs_config;
+        char file_name[30];
+        memcpy(file_name, &packet->data[IN_DATA_BYTE], 30);
+
+        strncat(file_name, ".txt", 4);
+        int32_t iErr = red_chdir("VOL0:/adcs");
+        if(iErr == -1){
+            sys_log(ERROR, "Error %d trying to change into adcs directory\r\n", red_errno);
+            status = ADCS_FILESYSTEM_FAIL;
+        }else{
+            int32_t file1 = red_open(file_name, RED_O_RDONLY);
+            if(file1 == -1){
+                sys_log(ERROR, "Error %d trying to open file %s\r\n", red_errno, file_name);
+                status = ADCS_FILE_DNE;
+            }else{
```

Comment:
Usually instead of having these big else clauses, the check for failure just returns